spec/forms/link_case/confirmation_form_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
+
+RSpec.describe LinkCase::ConfirmationForm, type: :form do
+  subject(:instance) { described_class.new(params) }
+
+  let(:params) do
+    {
+      model: linked_application,
+      link_type_code:,
+    }
+  end
+
+  let(:legal_aid_application) { create(:legal_aid_application) }
+  let(:linked_application) { create(:linked_application, lead_application_id: source_application.id, associated_application_id: legal_aid_application.id) }
+
+  let(:source_application) do
+    create(:legal_aid_application,
+           application_ref: "L-TVH-U0T",
+           provider: legal_aid_application.provider)
+  end
+
+  describe "#save" do
+    subject(:call_save) { instance.save }
+
+    context "with family link chosen" do
+      let(:link_type_code) { "FC_LEAD" }
+
+      it "updates the link_type_code" do
+        expect { call_save }.to change(linked_application, :link_type_code).from(nil).to("FC_LEAD")
+      end
+    end
+
+    context "with legal link chosen" do
+      let(:link_type_code) { "LEGAL" }
+
+      it "updates the link_type_code" do
+        expect { call_save }.to change(linked_application, :link_type_code).from(nil).to("LEGAL")
+      end
+    end
+
+    context "with no chosen" do
+      let(:link_type_code) { "false" }
+
+      it "does not update the link_type_code" do
+        expect { call_save }.not_to change(linked_application, :link_type_code).from(nil)
+      end
```

Comment:
[ditto](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/5933/files#r1399292688) but for update to nil

Reply:
plus we should have an example to exercise the destruction of the linked_application model instance.

```ruby
     it "does destroys the linked_application" do
        expect { call_save }.to change(LinkedApplication, :count).by(-1)
        # AND/OR
        call_save
        expect { linked_application.reload }.to raise_error(ActiveRecord::RecordNotFound)
      end
```

[examples](https://stackoverflow.com/questions/66108221/rspec-how-to-verify-if-a-record-has-been-deleted)